open-banking-accelerator/components/consent-management/com.wso2.openbanking.accelerator.consent.extensions/src/main/java/com/wso2/openbanking/accelerator/consent/extensions/manage/validator/VRPConsentRequestValidator.java

```diff
@@ -0,0 +1,505 @@
+        return value instanceof JSONArray;
+    }
+
+
+    /**
+     * Checks whether the given string represents a valid date-time in ISO-8601 format.
+     * <p>
+     * This method uses the ISO_DATE_TIME formatter to parse the provided date-time string.
+     * It returns true if the parsing is successful, indicating that the string is in the correct
+     * ISO-8601 date-time format. Otherwise, it returns false.
+     *
+     * @param value The string to be checked for ISO-8601 date-time format validity.
+     * @return
+     */
+    public static boolean isValidISODateTimeFormat(String value) {
+
+        final DateTimeFormatter dateTimeFormat = DateTimeFormatter.ISO_DATE_TIME;
+        try {
+            dateTimeFormat.parse(value);
+            return true;
+        } catch (DateTimeParseException e) {
+            return false;
+        }
+    }
+
+    /**
+     * Method to validate variable recurring payment periodic limits.
+     * This method validates the periodic limits specified in the control parameters for variable recurring payments.
+     * It checks if the provided JSON array of periodic limits is valid and then iterates through each limit
+     * to ensure that required fields such as amount, currency, period alignment, and period type are present and
+     * meet the specified criteria.
+     *
+     * @param controlParameters Initiation Object containing periodic limits
+     * @return validation response object indicating whether the provided periodic limits are valid
+     */
+    public static JSONObject validatePeriodicLimits(JSONObject controlParameters) {
+        JSONObject validationResponse = new JSONObject();
+
+        // Retrieve the periodic limits from the control parameters
+        Object periodicLimit = controlParameters.get(ConsentExtensionConstants.PERIODIC_LIMITS);
+
+        // Check if the control parameter is a valid JSON array
+        if (!isValidJSONArray(periodicLimit)) {
+            return ConsentManageUtil.getValidationResponse(ErrorConstants.PAYLOAD_FORMAT_ERROR_PERIODIC_LIMITS,
+                    ErrorConstants.INVALID_PARAMETER_PERIODIC_LIMITS,
+                    ErrorConstants.PATH_PERIOD_LIMIT);
+        }
+
+        // Check if the periodic limits key is present
+        if (controlParameters.containsKey(ConsentExtensionConstants.PERIODIC_LIMITS)) {
+
+            // Retrieve the JSON array of periodic limits
+            JSONArray periodicLimits = (JSONArray) controlParameters.get(ConsentExtensionConstants.PERIODIC_LIMITS);
+
+            // Iterate through each periodic limit
+            Iterator<Object> periodicLimitIterator = periodicLimits.iterator();
+
+            while (periodicLimitIterator.hasNext()) {
+                JSONObject limit = (JSONObject) periodicLimitIterator.next();
+
+                // Retrieve values for validation
+                Object amount = limit.getAsString(ConsentExtensionConstants.AMOUNT);
```

Comment:
Applicable for all the places
```suggestion
                String amount = limit.getAsString(ConsentExtensionConstants.AMOUNT);
```